### PR TITLE
Update readme and fix some typos

### PR DESCRIPTION
**Pull request by lovettchris**
Update the readme to match the current implementation and fix some typos in the code and remove dead code from the `old` folder.

### CODE REVIEW

File: README.md
Comment (lovettchris) [reply README.md]: fixed.

File: README.md
Comment (lovettchris) [reply README.md]: fixed.

File: README.md
Comment (lovettchris) [reply README.md:51]: Added CHANGELOG.md which seems to be the convention.

File: README.md
Comment (Vtec234) [inline README.md:8]: Is the all-`1` numbering intentional? It does display correctly (1,2,3,..) on GitHub.

Comment (Vtec234) [reply README.md:8]:
```suggestion
1. Open a folder containing Lean code using VS Code.  It is recommended that you open a folder that is configured using [Lake](https://github.com/leanprover/lake) which is a Lean package configuration tool.
```

File: README.md
Comment (Vtec234) [inline README.md:80]: If I'm not mistaken, this is just VSCode and not Lean specific.

File: README.md
Comment (Vtec234) [inline README.md:106]:
```suggestion
(1). The Infoview will activate automatically when a Lean file is opened, but you can also reopen it any time using the icon in the top right of the text editor window or the <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>P</kbd> `Lean 4: Infoview: Display Goal` command or the key that is bound to the command, which is <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Enter</kbd> by default.)
```

File: README.md
Comment (Vtec234) [inline README.md:124]:
```suggestion
(8). The "All Messages" widget can be expanded by clicking on it (or hitting the keybind for `lean4.displayList`, which is <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Enter</kbd> by default).
```

**Comment by Vtec234** [comment]
This is great, thanks so much for writing this! A few minor comments for now. I will give it a closer read later but overall the text looks good.

Regarding the `.png`s in `media/`, some of them can probably be deleted as we are now using the codicons font instead of embedded `.png`s.

File: README.md
Comment (lovettchris) [reply README.md]: Fixed and yes the "1." numbering trick makes it easier to re-order items in the list - the markdown processor numbers the items correctly in the generated HTML.  The vscode markdown viewer also does this correctly.

File: README.md
Comment (lovettchris) [reply README.md]: Ok, I updated the list with more that I think are lean specific because they are triggered by abbreviations.

**Comment by lovettchris** [comment]
> Regarding the `.png`s in `media/`, some of them can probably be deleted as we are now using the codicons font instead of embedded `.png`s.

I backed off using the codicons in the markdown since github doesn't support it.  But I deleted a few more old/media files that were redundant.

File: lean4-infoview/src/infoview/editorConnection.ts
Comment (Vtec234) [inline lean4-infoview/src/infoview/editorConnection.ts:35]: Leftover?

Comment (lovettchris) [reply lean4-infoview/src/infoview/editorConnection.ts]: woops. nice catch. Fixed.

**Review by Vtec234 — APPROVED** [review]
Thanks again!

**Comment by lovettchris** [comment]
> Thanks again!

you're welcome, hey what is the check-in process, is 1 approval enough, or do I also need to wait for Gabriel's approval?

**Comment by Vtec234** [comment]
I think usually we would wait a day or so to see if Gabriel or others have further comments, and then merge.

**Comment by gebner** [comment]
> do I also need to wait for Gabriel's approval?

I trust Wojciech's judgement on merging PRs.  Practically speaking, you need to wait for me anyhow since I'm the only one who can deploy the extension atm.

**Pull request merged**
